### ISSUE TITLE
Cannot read property 'getLineAndCharacterOfPosition' of undefined

### ISSUE THREAD

**Issue by nidu** [TypeStrong/fork-ts-checker-webpack-plugin#80]
Hello. Just migrated to webpack 3 and trying to setup fast TS compilation, but stumbled on following error. Problem occurs for TS 2.6.x but not for e.g. 2.1.4. Any hints on what can be the cause? Maybe some broken dependency? If i proceed working and save files same error appears again and again.

Environment:

Windows 7x64
webpack 3.10.0
node 6.11.0
npm 3.10.7
typescript 2.6.2
fork-ts-checker-webpack-plugin 0.2.9
ts-loader 3.2.0

Plugin is created without options and here's ts-loader config:

```
{ 
  test: /\.tsx?$/, 
  loader: 'ts-loader', 
  options: {
    // disable type checker - we will use it in fork plugin
    transpileOnly: true,
    happyPackMode: true
  } 
}
```

Error:

```
Starting type checking service...
Using 1 worker with 2048MB memory limit
 67% building modules 982/1020 modules 38 active ...s\user-monitor3\UserStore.tsc:\nidu\dev\juno\juno\node_modules\fork-ts-checker-webpack-plugin\lib\service.js:22
        throw error;
        ^

TypeError: Cannot read property 'getLineAndCharacterOfPosition' of undefined
    at NormalizedMessage.createFromDiagnostic (c:\nidu\dev\juno\juno\node_modules\fork-ts-checker-webpack-plugin\lib\NormalizedMessage.js:15:39)
    at Array.map (native)
    at IncrementalChecker.getDiagnostics (c:\nidu\dev\juno\juno\node_modules\fork-ts-checker-webpack-plugin\lib\IncrementalChecker.js:125:58)
    at run (c:\nidu\dev\juno\juno\node_modules\fork-ts-checker-webpack-plugin\lib\service.js:13:31)
    at process.<anonymous> (c:\nidu\dev\juno\juno\node_modules\fork-ts-checker-webpack-plugin\lib\service.js:38:5)
    at emitTwo (events.js:106:13)
    at process.emit (events.js:191:7)
    at process.nextTick (internal/child_process.js:758:12)
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
 94% asset optimization
```

**Comment by johnnyreilly** [comment]
I'd suggest clearing your node modules and reinstalling - btw you didn't share your plugin setup..

**Comment by nidu** [comment]
@johnnyreilly Thanks, it helped!

**Comment by nidu** [comment]
Unfortunately i was wrong. Error just didn't appear until i resolve webpack errors. Plugin configuration is default - just `new ForkTsCheckerWebpackPlugin()`. Maybe there's some dependency that probably may be the reason?

EDIT: Error starts with typescript@2.4.0, 2.3.2 doesn't show this error.

**Comment by nidu** [comment]
According to [this](https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API#incremental-build-support-using-the-language-services) (function `logErrors`) `diagnostic.file` can be nil.

Wrapping code inside `NormalizedMessage.createFromDiagnostic` in `if (diagnostic.file)` and printing diagnostic i've got in console:

```
 66% building modules 862/921 modules 59 active ...ct\lib\ReactTransitionChildMapping.js###DIAG { file: undefined,
  start: undefined,
  length: undefined,
  messageText: 'Cannot find global value \'Promise\'.',
  category: 1,
  code: 2468 }
 95% emittingERROR in undefined(undefined,undefined):
TS2468: Cannot find global value 'Promise'.
```

So maybe reading `diagnostic.file` attributes should be wrapped in `if (diagnostic.file)`?

**Comment by johnnyreilly** [comment]
Thanks for digging into this @nidu.  You're seeing:

```
TS2468: Cannot find global value 'Promise'.
```

Are you running in a super old version of node?  

It would be useful if you could provide a minimal reproduction of the issue; it's hard to know if this is a configuration problem or a bug with fork-ts-checker-webpack-plugin.

**Comment by nidu** [comment]
@johnnyreilly environment is as following:

Windows 7x64
webpack 3.10.0
node 6.11.0
npm 3.10.7
typescript 2.6.2
fork-ts-checker-webpack-plugin 0.2.9
ts-loader 3.2.0

I resolved the error by adding `es6-promise` to the project so this particular error was mine. But at the same time i think checker should print error and errors like this can occur. Moreover according to example from my link above `diagnostic.file` can be undefined so i suppose nil check is worth adding.

**Comment by johnnyreilly** [comment]
Yeah that sounds reasonable. If you get a chance to submit a PR I'm sure @piotr-oles would be appreciative

**Comment by piotr-oles** [comment]
PR merged, will be released as v0.2.10 :)
